core/templates/services/search.service.ts

```diff
@@ -16,215 +16,234 @@
         }
+      });
 
-        return decodeURIComponent(querySegments[0]);
-      },
-      getCurrentUrlQueryString: function() {
-        return this.getSearchUrlQueryString(
-          _lastQuery, _lastSelectedCategories, _lastSelectedLanguageCodes);
-      },
-      loadMoreData: function(successCallback, failureCallback) {
-        // If a new query is still being sent, or the end of the page has been
-        // reached, do not fetch more results.
-        if (_isCurrentlyFetchingResults || hasReachedEndOfPage()) {
-          failureCallback(hasReachedEndOfPage());
-          return;
-        }
+    if (successCallback) {
+      successCallback();
+    }
+  }
 
-        var queryUrl = getQueryUrl(this.getCurrentUrlQueryString());
+  isSearchInProgress(): boolean {
+    return this.numSearchesInProgress > 0;
+  }
 
-        if (_searchCursor) {
-          queryUrl += '&cursor=' + _searchCursor;
-        }
+  /**
+   * Takes in the url search component as an argument and the
+   * selectionDetails. It will update selectionDetails with the relevant
+   * fields that were extracted from the url.
+   * @returns the unencoded search query string.
+  */
+  updateSearchFieldsBasedOnUrlQuery(
+      urlComponent:string, selectionDetails:SelectionDetails): string {
+    var urlQuery = urlComponent.substring('?q='.length);
+    // The following will split the urlQuery into 3 components:
+    // 1. query
+    // 2. categories (optional)
+    // 3. language codes (default to 'en').
+    var querySegments = urlQuery.split('&');
+
+    for (var i = 1; i < querySegments.length; i++) {
+      urlComponent = decodeURIComponent(querySegments[i]);
+
+      var itemsType = null;
+      if (urlComponent.indexOf('category') === 0) {
+        itemsType = 'categories';
+      } else if (urlComponent.indexOf('language_code') === 0) {
+        itemsType = 'languageCodes';
+      } else {
+        continue;
+      }
+
+      try {
+        this.updateSearchFields(itemsType, urlComponent, selectionDetails);
+      } catch (error) {
+        selectionDetails[itemsType].selections = {};
+      }
+    }
+
+    return decodeURIComponent(querySegments[0]);
+  }
+
+  getCurrentUrlQueryString():string {
+    return this.getSearchUrlQueryString(
+      this.lastQuery,
+      this.lastSelectedCategories,
+      this.lastSelectedLanguageCodes
+    );
+  }
 
-        _isCurrentlyFetchingResults = true;
-        $http.get(queryUrl).then(function(response) {
-          _searchCursor = response.data.search_cursor;
-          _isCurrentlyFetchingResults = false;
-
-          if (successCallback) {
-            successCallback(response.data, hasReachedEndOfPage());
-          }
-        });
-      },
-
-      get onSearchBarLoaded() {
-        return _searchBarLoadedEventEmitter;
-      },
-      get onInitialSearchResultsLoaded() {
-        return _initialSearchResultsLoadedEventEmitter;
+  loadMoreData(
+      successCallback:(SearchResponseData, boolean)=>void,
```

Comment:
Space before and after `=>`
```suggestion
      successCallback: (SearchResponseData, boolean) => void,
```

Reply:
done